### PR TITLE
move: source service config

### PR DESCRIPTION
**Pull request by rvantonder**
## Description 

This adds a basic config file to initialize the server with. It takes a list of `Packages`, where `Packages` comprise a repository and one or more paths to Move package in that repository. 

## Test Plan 

Added a basic test to parse a config file.

---
If your changes are not user-facing and not a breaking change, you can skip the following section. Otherwise, please indicate what changed, and then add to the Release Notes section as highlighted during the release process.

### Type of Change (Check all that apply)

- [ ] protocol change
- [ ] user-visible impact
- [ ] breaking change for a client SDKs
- [ ] breaking change for FNs (FN binary must upgrade)
- [ ] breaking change for validators or node operators (must upgrade binaries)
- [ ] breaking change for on-chain data layout
- [ ] necessitate either a data wipe or data migration

### CODE REVIEW

File: crates/sui-source-validation-service/src/lib.rs [hunk 1]
```diff
@@ -37,7 +52,26 @@ pub async fn verify_package(
...
+    toml::from_str(&contents).map_err(anyhow::Error::from)
+}
+
+pub async fn clone_repositories(config: &Config) -> anyhow::Result<()> {
+    for p in &config.packages {
+        let _ = p.repository;
+        let _ = p.paths;
+    }
+    Ok(())
+}
```
Comment (rvantonder) [inline crates/sui-source-validation-service/src/lib.rs:66]: Added to remove `unused` warning, will be implemented in following PR

File: crates/sui-source-validation-service/tests/tests.rs [hunk 2]
```diff
@@ -35,3 +36,24 @@ async fn test_api_route() -> anyhow::Result<()> {
...
+        "crates/sui-framework/packages/deepbook",
+        "crates/sui-framework/packages/move-stdlib",
+        "crates/sui-framework/packages/sui-framework",
+        "crates/sui-framework/packages/sui-system",
+    ]
+"#;
+
+    let config: Config = toml::from_str(config).unwrap();
+    let expect = expect![
+        r#"Config { packages: [Packages { repository: "https://github.com/mystenlabs/sui", paths: ["crates/sui-framework/packages/deepbook", "crates/sui-framework/packages/move-stdlib", "crates/sui-framework/packages/sui-framework", "crates/sui-framework/packages/sui-system"] }] }"#
```
Comment (rvantonder) [inline crates/sui-source-validation-service/tests/tests.rs:55]: Not sure if there's a prettier way to output this than writing a `Display`, which I figured to do later

Comment (rvantonder) [reply crates/sui-source-validation-service/tests/tests.rs]: Fixed this with `:#?`

File: crates/sui-source-validation-service/src/lib.rs (hunk 1 above)
Comment (amnn) [reply crates/sui-source-validation-service/src/lib.rs:70]: You could probably just prefix `config` with an underscore, i.e. `_config`.

File: crates/sui-source-validation-service/src/lib.rs [hunk 3]
```diff
@@ -41,7 +56,26 @@ pub async fn verify_package(
         .map_err(anyhow::Error::from)
 }
 
-pub async fn initialize(
+pub fn parse_config(config_path: impl AsRef<Path>) -> anyhow::Result<Config> {
+    let contents = fs::read_to_string(config_path)?;
+    toml::from_str(&contents).map_err(anyhow::Error::from)
```
Comment (amnn) [inline crates/sui-source-validation-service/src/lib.rs:61]: ...come to think of it, I think `?` already does type conversion on the error, so the idiomatic way to do this is as follows:

```suggestion
    Ok(toml::from_str(&contents)?)
```

**Review by amnn — APPROVED** [review]
lgtm!

File: crates/sui-source-validation-service/src/lib.rs (hunk 3 above)
Comment (rvantonder) [reply crates/sui-source-validation-service/src/lib.rs:61]: Oh I like that! Good catch!

**Pull request merged**